### ISSUE TITLE
Do we build contract bundles that generalize to a given use case? Or just greenlight custom contracts built and submitted by adopters? 

### ISSUE THREAD

**Issue by arjunhassard** [nucypher/nucypher#3180]
_Build generalized contract bundle_
Pros: 
- We are in a better position to build these given our understanding of the underlying schemes, network economics, etc. It may actually be more work to hand-hold and correct attempts by external developers. 
Cons:
- We will have to dedicate significant time to each bundle offering. 
- We will inevitably lose business by (i) not choosing the right granularity of bundle and (ii) not having bundles ready in time.  

_Approve externally developed contracts_
Pros: 
- Avoids adopters waiting for us to complete a contract bundle designed for their domain. In theory we can quickly onboard a new adopter if they have followed our guidelines very carefully. 
Cons: 
- It doesn't make a lot of sense for adopters to implement their own fee structures / payment gates  
- It will still take a lot of effort and time to (1) guide the development of these contracts, (2) analyze the fee structure vs. a given user base's expected behavior, (3) negotiate / go back-and-forth with adopters over their proposed bundles

**Comment by ghardin1314** [comment]
> Cons:
> 
> * It doesn't make a lot of sense for adopters to implement their own fee structures / payment gates

I disagree with this statement. I think this is the core of how an adopter makes their own business modal on top of TACo/CBD. By having a standard fee structure for TACo specific actions (Ritual creation/refresh, cohort lifespan), an adopter can then place their own payments on top for app specific actions (upgradeable conditions, pay per encryption, ...). This makes it sustainable for both parties involved. 

If all payments for an app are just flowing through directly to TACo and their is not value capture at the App level, there is no incentive for an app to really drive user growth other than being a Public Good
